src/python/pants/option/options.py

```diff
@@ -73,16 +74,16 @@ class DuplicateScopeError(Exception):
         """More than one registration occurred for the same scope."""
 
     @classmethod
-    def complete_scopes(cls, scope_infos: Iterable[ScopeInfo]) -> Set[ScopeInfo]:
+    def complete_scopes(cls, scope_infos: Iterable[ScopeInfo]) -> FrozenOrderedSet[ScopeInfo]:
```

Comment:
Sometimes it's fine to return a `Set`, but generally `FrozenOrderedSet` is a much safer default. Especially with the engine, we want stable iteration order for caching.